docs/web/authentication.md

```diff
@@ -402,73 +442,73 @@ CodeChecker also supports OAuth-based authentication. The `authentication.method
           * `username` - Company's signum will be the user's identifier.
           * `email` - an email associated with this Microsoft account will be used as user's identifier.
 
-    ### 🔧 Example: OAuth Configuration for GitHub
-    ~~~{.json}
+    ### 🔧 Example: OAuth Configuration using templates
+    
+    ```jsonc
     "github": {
       "enabled": false,
       "client_id": "<ExampleClientID>",
       "client_secret": "<ExampleClientSecret>",
-      "authorization_url": "https://github.com/login/oauth/authorize",
-      "callback_url": "https://<server_host>/login/OAuthLogin/github",
-      "token_url": "https://github.com/login/oauth/access_token",
-      "user_info_url": "https://api.github.com/user",
-      "user_emails_url": "https://api.github.com/user/emails",
-      "scope": "user:email",
-      "user_info_mapping": {
+      "template": "github/v1",
+      "user_info_mapping": { // optional
         "username": "username"
       }
     }
-    ~~~
-    ### 🔧 Example: OAuth Configuration for Google
-    ~~~{.json}
-    "google": {
+    ```
+
+    ### 🔧 Example: OAuth Configuration for Microsoft
+
+    ```jsonc
+    "microsoft": {
       "enabled": false,
       "client_id": "<ExampleClientID>",
       "client_secret": "<ExampleClientSecret>",
-      "authorization_url": "https://accounts.google.com/o/oauth2/auth",
-      "callback_url": "https://<server_host>/login/OAuthLogin/google",
-      "token_url": "https://accounts.google.com/o/oauth2/token",
-      "user_info_url": "https://www.googleapis.com/oauth2/v1/userinfo",
-      "scope": "openid email profile",
-      "user_info_mapping": {
+      "template": "ms_entra/v2.0",
+      "variables": {
+        "tenant_id": "common" // replace with your own tenant ID, if not using multi-tenant mode
+      },
+      "user_info_mapping": { // optional
         "username": "email"
       }
     }
-    ~~~
-    ### 🔧 Example: OAuth Configuration for Microsoft
-    ~~~{.json}
-    "microsoft": {
+    ```
+
+    ### 🔧 Example: OAuth Configuration without templates
+
+    ```jsonc
+    "google": {
       "enabled": false,
       "client_id": "<ExampleClientID>",
       "client_secret": "<ExampleClientSecret>",
-      "authorization_url": "https://login.microsoftonline.com/<tenant-id>/oauth2/v2.0/authorize",
-      "callback_url": "https://<server_host>/login/OAuthLogin/microsoft",
-      "token_url": "https://login.microsoftonline.com/<tenant-id>/oauth2/v2.0/token",
-      "user_groups_url": "https://graph.microsoft.com/v1.0/me/memberOf",
-      "jwks_url": "https://login.microsoftonline.com/<tenant-id>/discovery/v2.0/keys",
-      "user_info_url": "https://graph.microsoft.com/v1.0/me",
-      "scope": "User.Read email profile openid offline_access",
+      "template": "google/v1", // still needed for the response handling logic
+      "authorization_url": "https://accounts.google.com/o/oauth2/auth",
+      "callback_url": "https://<server_host>/login/OAuthLogin/google",
```

Comment:
can the user use variables also here or only in the templates?

Reply:
They can, correct. Added example.